frame.go

```diff
@@ -0,0 +1,70 @@
+package errtrace
+
+import (
+	"fmt"
+	"runtime"
+)
+
+// Frame is a single frame in an error trace
+// identifying a site where an error was wrapped.
+type Frame struct {
+	// Func is the fully qualified function name
+	// inside which the error was wrapped.
+	Func string
+
+	// File is the file inside which the function is defined.
+	File string
+
+	// Line is the line number inside the file
+	// where the error was wrapped.
+	Line int
+}
+
+func (f Frame) String() string {
+	return fmt.Sprintf("%s (%s:%d)", f.Func, f.File, f.Line)
+}
+
+// UnwrapFrame unwraps the outermost frame from the given error,
+// returning it and the inner error.
+// If the outermost error is not an errtrace-wrapped error,
+// UnwrapFrame returns (Frame{}, err, false).
+//
+// You can use this for structured access to trace information.
+// For example:
+//
+//	err := // ..
+//	var frames []Frame
+//	for {
+//		frame, err, ok := UnwrapFrame(err)
+//		if !ok {
+//			break
+//		}
+//		frames = append(frames, frame)
+//	}
+//
+// Note that the loop like the above will stop
+// when it encounters an error that wasn't wrapped
+// with errtrace.Wrap or its friends.
+// A fully complete version will also want to handle
+// errors that wrap other errors in a different way,
+// and multi-errors where each error has its own trace.
+func UnwrapFrame(err error) (frame Frame, inner error, ok bool) { //nolint:revive // error is intentionally middle return
+	e, ok := err.(*errTrace)
+	if !ok {
+		return Frame{}, err, false
+	}
+
+	frames := runtime.CallersFrames([]uintptr{e.pc})
```

Comment:
this is a good place to start, just wanted to think about a couple of things:
 * should we return a frame with the func/line as frields, or should we instead return a type that wraps the `pc` and lazily pulls the frame as needed.
 * I wonder if there's a performance benefit in `CallersFrames` when multiple pcs are passed.

I don't think this function is expected to be used in any performance-sensitive paths, so don't think it's worth optimizing for it yet.

In the future, if batching is faster, we may want to switch to doing that internally (in which case we couldn't use this function), which is still OK.

Reply:
Yeah, I was unsure about expose Frame with decoded information or just a hidden pc that loads the frame lazily. 
I started with exposed frame since that's what we had for traceFrame, and it's the most "here's just data,"
but I think lazy decoding leaves room for frame iteration in the future, should we want that.

> I wonder if there's a performance benefit in CallersFrames when multiple pcs are passed.

Doesn't look like it. https://cs.opensource.google/go/go/+/refs/tags/go1.22.2:src/runtime/symtab.go;l=91
You pay for one caller at a time, although it pays for two frames for the initial `Next()` call if there are multiple frames.

Reply:
Re: lazy vs eager, I'm leaning towards the eager approach like you have it for simplicity. That said, what do you think our [own Frame type vs exposing the runtime type](https://github.com/bracesdev/errtrace/pull/102#issuecomment-2057222444)?

Re: performance benefit of batching,

I wrote a benchmark for an error with ~6 frames and compared a for loop that builds up `[]uintptr` of PCs, and uses `CallerFrames` to then generate a `[]string` of function names, and compared doing the same using `UnwrapFrame`. The time is similar, though it is more allocation-friendly to batch `CallerFrames` calls,
```
BenchmarkUnwrapBatch 	 2007859	       578.3 ns/op	     464 B/op	       3 allocs/op
BenchmarkUnwrapFunc  	 1301568	       924.6 ns/op	    1552 B/op	      13 allocs/op
```

So if we want to optimize the tree building, batching may be worth it, but I don't think that's worth changing the API for.

Reply:
> * should we return a frame with the func/line as frields, or should we instead return a type that wraps the `pc` and lazily pulls the frame as needed.

Are there any plans to _not_ use PCs in the errtrace implementation?

In https://go.dev/issue/63358, a few people have said that `[]uintptr` seems like a good way to represent a trace, though https://go.dev/issue/60873 is more appropriate for that discussion.

Personally, I do not mind creating `runtime.Frame`s myself from one or more `uintptr`. 🤔 I think I just want to read the unexported `pc` field.

Reply:
> Are there any plans to not use PCs in the errtrace implementation?

No, no plans to use non-PC sources of information.
`[]uintptr` for a trace seems like a good idea, but those discussions do not seem close to resolving.
I think we could expose these frames as-is for now, and add a means of exposing the `[]uintptr` in the future.
That may happen anyway if we implement #104.